src/models/StyledComponent.js

```diff
@@ -23,17 +23,30 @@ import ServerStyleSheet from './ServerStyleSheet'
 const STATIC_EXECUTION_CONTEXT = {}
 
 export default (ComponentStyle: Function, constructWithOptions: Function) => {
-  /* We depend on components having unique IDs */
   const identifiers = {}
+
+  /* We depend on components having unique IDs */
   const generateId = (_displayName: string, parentComponentId: string) => {
     const displayName =
       typeof _displayName !== 'string' ? 'sc' : escape(_displayName)
 
-    const nr = (identifiers[displayName] || 0) + 1
-    identifiers[displayName] = nr
+    let componentId
+
+    /**
+     * only fall back to hashing the component injection order if
+     * a proper displayName isn't provided by the babel plugin
+     */
+    if (!_displayName) {
```

Comment:
Why are we using `_displayName` vs. `displayName` in the line below? Can we clean this up?

Reply:
Because `displayName` gets normalized to the string `"sc"` which won't be falsy.

Reply:
I guess I could do `if (displayName === 'sc')` but it's possible that a user could actually name something `"sc"` right?